evap/student/tests/test_views.py

```diff
@@ -17,64 +21,127 @@ def setUp(self):
 
 
 @override_settings(INSTITUTION_EMAIL_DOMAINS=["example.com"])
-class TestVoteView(WebTest):
-    fixtures = ['minimal_test_data']
+class TestVoteView(ViewTest):
+    url = '/student/vote/'
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.voting_user1 = mommy.make(UserProfile)
+        cls.voting_user2 = mommy.make(UserProfile)
+        cls.contributor1 = mommy.make(UserProfile)
+        cls.contributor2 = mommy.make(UserProfile)
+
+        cls.course = mommy.make(Course, pk=5, participants=[cls.voting_user1, cls.voting_user2, cls.contributor1], state="in_evaluation")
+
+        cls.general_questionnaire = mommy.make(Questionnaire)
+        cls.contributor_questionnaire = mommy.make(Questionnaire)
+
+        cls.contributor_text_question = mommy.make(Question, questionnaire=cls.contributor_questionnaire, type="T")
+        cls.contributor_likert_question = mommy.make(Question, questionnaire=cls.contributor_questionnaire, type="L")
+        cls.general_text_question = mommy.make(Question, questionnaire=cls.general_questionnaire, type="T")
+        cls.general_likert_question = mommy.make(Question, questionnaire=cls.general_questionnaire, type="L")
+        cls.general_grade_question = mommy.make(Question, questionnaire=cls.general_questionnaire, type="G")
+
+        cls.contribution1 = mommy.make(Contribution, contributor=cls.contributor1, questionnaires=[cls.contributor_questionnaire],
+                                       course=cls.course)
+        cls.contribution2 = mommy.make(Contribution, contributor=cls.contributor2, questionnaires=[cls.contributor_questionnaire],
+                                       course=cls.course)
+
+        cls.course.general_contribution.questionnaires.set([cls.general_questionnaire])
+
+        cls.url = cls.url + str(cls.course.pk)
```

Comment:
just specify the course pk in the mommy call and write that into the url directly. not very pretty, but all the other tests do it this way too.